# **QC Solutions for SARS-CoV-2 Genomic Analysis**

**PHA4GE Bioinformatics Pipelines &amp; Visualization Working Group** <br/>
Libuit KG, Lunn S, Carleton H, Khan W, Kanwar S, van Heusden P, Amrosio F, Lemmer D, Mboowa G, Macori G, Southgate J 

<details>
 <summary> Document Change Log</summary>
 
- {date}:
  - First draft published
</details>


# Overview

Next-generation sequencing (NGS) has expanded the approach of genomic analysis for pathogen surveillance systems. The demand for NGS continues to grow, with the need for high throughput, lower costs, and better quality of data. 

However, the quality of NGS sequencing data can be affected by library preparation and sequencing processes, systematic variation in quality scores across sequence reads, biases in sequencing due to base composition, and less-than optimal library fragment sizes and indexes. Such factors can negatively impact the quality of raw sequencing data for downstream analyses. 

In an attempt to assist with quality control (QC) measures, the bioinformatics pipeline and visualization working group of the Public Health Alliance for Genomic Epidemiology (PHA4GE) has drafted this living document to help define the QC challenges for SC2 genomic analysis and suggest a QC systems solutions to address them.

Please note that the QC guidelines in this document are simply an attempt to highlight the most accessible solutions **as per the opinions of our working group** and in no way represent a comprehensive system for QC guidance and bioinformatic solutions. If this document fails to include a valuable public health resource or in some way mischaracterizes a resource mentioned, we encourage community collaboration through pull-requests and/or raised GitHub issues. 

## Contents
- [Process Control For Bioinformatics QC Checkpoints](#process-control-for-bioinformatics-qc-checkpoints)
- [QC Acceptance Criteria](#qc-acceptance-creiteria)
  - [PHA4GE Suggested Thresholds](#pha4ge-suggested-thresholds)
- [QC Metric Definitions](#qc-metric-definitions)
  - [Read QC Metrics](#read-qc-metrics)
  - [Alignment QC Metrics](#alignment-qc-metrics)
  - [Consensus Assembly QC Metrics](#consensus-assembly-qc-metrics)
- [Additional QC Resources and Materials](#additional-qc-resrouces-and-materials)

# Process Control For Bioinformatics QC Checkpoints
The focus of this document is on the quality control (QC) of tiled amplicon sequencing--through the Artic V3 protocol, for example--a common method for generating SC2 sequencing data. These sequencing experiments generate thousands of amplicon reads that represent fragments of the original SC2 genome present in a sample and--as discussed in this working group's [Bioinformatics Solutions for SARS-CoV-2 Genomic Analysis Guidance Document](../docs/bioinfo-solutions.md)--assembling a contiguous SC2 genome from these amplicon read data is a critical step in providing insight from sequenced samples.

Throughout this process, quality control checkpoints should be conducted at different stages of bioinformatics analysis, including QC of raw read data, pre-processing stages (trimming and filtering), and alignment/assembly.

<p align="center">
  <img src="./images/pha4ge_sc2_qc_workflow.png" width=10800" class="center">
</p>

In this context, raw read data refers to the fastq read files generated by the NGS platform and processed reads are read files that have had adapter sequences removed, trimmed based on size and quality, and dehosted. Alignment QC refers to the examination of the BAM or VCF files generated during the consensus genome assembly process and Consensus Assembly QC refers to an assessment of the fasta assembly file itself.

Future updates to this document will include QC guidance for SARS-CoV-2 genomic epidemiology analysis and wastewater sequencing data.
									    
# QC Acceptance Criteria

When performing QC checks on SARS-CoV-2 genomic data, it can be helpful to establsh acceptance thresholds to determine how and when data will be reported and utilized to inform public health decision-making. Below are this working group's suggested QC thresholds for SARS-CoV-2 genomic data as well as various resources and metric definitions to assist in public health laboratories implmenting SARS-CoV-2 sequencing and analysis protocols. 
									    
## PHA4GE Suggested Thresholds

<table>
  <tbody>
    <tr>
      <th align="center" colspan=2 background-color="#F00000">Read QC Metrics</th>
    </tr>
    <tr>
      <td rowspan=1 height="5px"><a href="#number-of-reads">Number of Reads </a></td>
      <td rowspan=1>Protocol dependent, (e.g. 100,000 reads from Artic Amplicons sequenced on Illumina MiSeq)</td>
    </tr>
    <tr>
      <td width="510px;" rowspan=1 "><a href="#percent-human-reads">Percent Human Reads</a></td>
      <td width="510px;" rowspan=1><20%</td>
    </tr>
    <tr>
      <th align="center" colspan=2>Alignment QC Metrics</th>
    </tr>
    <tr>
      <td rowspan=1 height="5px"><a href="#sequencing-depth">Average Read Depth</a></td>
      <td rowspan=1>≥100x </td>
    </tr>
    <tr>
      <td rowspan=1 height="5px"><a href="#percent-mapped-reads">Percent mapped reads to Wuhan reference genome</a></td>
      <td rowspan=1>≥65% </td>
    </tr>
    <tr>
      <td rowspan=1 height="5px"><a href="#coverage">Coverage at a Single Base to Make a Base Call</a></td>
      <td rowspan=1>≥50x </td>
    </tr>
    <tr>
      <td rowspan=1 height="5px"><a href="#percent-agreement">Percent Agreement</a></td>
      <td rowspan=1>80%</td>
    </tr>
    <tr>
      <td rowspan=1 height="5px"><a href="#average-base-quality-of-aligned-reads">Average base quality of aligned reads</a></td>
      <td rowspan=1>>15</td>
    </tr>
    <tr>
	    <th align="center" colspan=2><u>Assembly QC Metrics</u></th>
    </tr>
    <tr>
      <td rowspan=1 height="5px"><a href="#percent-reference-coverage">Percent reference coverage</a></td>
      <td rowspan=1>>83%</td>
    </tr>
    <tr>
      <td rowspan=1 height="5px"><a href="#number-of-ns">Number of Ns</a></td>
      <td rowspan=1><5,000bp</td>
    </tr>
    <tr>
      <td rowspan=1 height="5px"><a href="#assembly-length-unambiguous">Assembly length unambiguous</a></td>
      <td rowspan=1>>24,000bp</td>
    </tr>
    <tr>
      <td rowspan=1 height="5px"><a href="#ntc-percent-coverage">NTC percent coverage</a></td>
      <td rowspan=1><10%</td>
    </tr>
    <tr>
      <td rowspan=1 height="5px"><a href="#lineage-defining-mutations">Lineage defining mutations</a></td>
      <td rowspan=1>≥60%</td>
    </tr>
    <tr>
      <td rowspan=1 height="5px"><a href="#s-gene-coverage">S-gene coverage</a></td>
      <td rowspan=1>≥99%</td>
    </tr>
    <tr>
      <td rowspan=1 height="5px"><a href="#s-gene-frameshifts">S-gene frameshifts sequence</a></td>
      <td rowspan=1>0</td>
    </tr>
    <tr>
      <td rowspan=1 height="5px"><a href="#s-gene-ambiguous-bases">S-gene ambiguous bases</a></td>
      <td rowspan=1><10%</td>
    </tr>
  </tbody>
</table>
								    
									    
## QC Metric Definitions

### Read QC Metrics
Different sequencing platforms use different technologies to determine the nucleotide sequence of the genetic material that they are processing, but all of these technologies converge on the fastq file format. For example, Illumina uses a sequencing-by-synthesis approach which involves assembling copies of each read using fluorescently tagged nucleotides and taking high resolution pictures of each read as each nucleotide is added to the read. These images are then captured in binary base call (BCL) files, and BCL files are converted into fastq files using the bcl2fastq program. On the other hand, Oxford Nanopore Technologies sequencing platforms run single strands of nucleic acids through nano-scale protein pores. An electric current is run across the pore, and the changes in current are detected as each nucleotide passes through the pore. The raw electric signal is captured in the fast5 file format and converted into fastq file format using the basecalling program guppy. Due to the nature of these sequencing platforms there are different considerations when assessing the quality of the raw sequence data (the fastq files).

| Term                  | Definition                             |
| ---------------------- | --------------------------------------- |
| <h4>Reads</h4> | Fragments of sequence DNA base pairs that are generated during sequencing; also referred to as the raw data generated from a sequencing platform |
| <h4>Number of Reads</h4> | Count of reads generated in an NGS run|
| <h4>BCL Files</h4> | Raw image files produced by Illumina instruments, converted to fastq via bcl2fastq program |
| <h4>FAST5 Files</h4> | Raw electrical signal files produced by Oxford Nanopore Technologies sequencing equipment, converted to fastq via basecalling software (guppy is the current industry standard) |
| <h4>Basecalling</h4> | The computational process of translating raw electrical signal files (FAST5) or flowcell images (BCL) to nucleotide sequence <br/>[Performance of neural network basecalling tools for Oxford Nanopore sequencing](https://pubmed.ncbi.nlm.nih.gov/31234903/) |
| <h4>FASTQ Files</h4> | The common “raw” sequence files containing nucleotide sequences and their associated quality scores  <br/> &bull; The quality scores contained within a fastq file are encoded as ASCII characters so that they require one bit per score making the string of nucleotide sequences and the string of quality scores equal in length <br/> &bull; The quality score (Q Score) represents the probability of an accurate base assignment at the associated nucleotide position <br/> &bull; Q scores range from 0 to 40 and are mathematically equivalent to: <br>&nbsp;&nbsp;&nbsp;&nbsp; <pre> Q = -10log<sub>10</sub>P</pre> &bull; [Quality Scores for Next-Generation Sequencing - illumina](https://www.illumina.com/documents/products/technotes/technote_Q-Scores.pdf) <br/>&bull; [Measuring sequencing accuracy - illumina](https://emea.illumina.com/science/technology/next-generation-sequencing/plan-experiments/quality-scores.html) <br/> &bull; Q Scores for Illumina and ONT sequencing will differ dramatically <br>&nbsp;&nbsp;&nbsp;&nbsp; &bull; An excellent Illumina run will have an average Q Score of 27-30 <br>&nbsp;&nbsp;&nbsp;&nbsp; &bull; An excellent Nanopore run will have an average Q Score of 12-15 <br/> &bull; Low Q Scores indicate poor sequencing quality which will impact all downstream analyses | 
| <h4>Ambiguity / Mixed Sites</h4> | The percent of each read where the base called is ambiguous <br/> [IUPAC Codes](https://www.bioinformatics.org/sms/iupac.html) |
| <h4>Sequence GC Content</h4> | The GC content of reads should be normally distributed |
| <h4>Raw vs Processed Reads</h4> | It is typical for some reads to be removed during quality filtering. Based on the known characteristics of the sample, one should be able to predict a reasonable proportion of the reads to be removed.|
| <h4>Percent Human Reads</h4> | Percentage of human read data sequenced in an NGS run. |

### Alignment QC Metrics
Consensus-genome assembly approaches have been widely adopted for SARS-CoV-2 genomic analysis. In this approach, read data are aligned to a reference genome--usually [Wuhan-1 (MN908947.3)] (https://www.ncbi.nlm.nih.gov/nuccore/MN908947.3)--and each position in the alignment are assessed to determine the consensus basecall supported by the read data at each position. The alignments are captured in a BAM file that can be used to assess critical quality control metrics; additionally VCF files can be produced from an alignment to call variant positions relative to the reference genome--VCF files can also be inspected to assess quality of identified variant positions. 

| Term                  | Definition                             |
| ---------------------- | --------------------------------------- |
| <h4>Sequence Alignment</h4>  | A method of arranging nucleic acid (DNA/RNA) or protein sequences to identify regions of similarity or conservation that may be of function, structural, or evolutionary relationships. Pairwise sequence alignment consists of two sequences whereas multiple sequence alignment consists of more than three sequences |
|<h4>Sequencing Depth</h4> | The number of reads that cover a particular nucleotide, section/amplicon of the genome, or average across the reference sequence<br/>  &bull; Ideally a min depth of 10X for Illumina or 20X for Nanopore would be reached<br/>&bull; Uniform depth of coverage is better<br/> &bull; Nonuniform depth may be indicative of differential amplification of amplicons, or amplicon dropout<br/> &nbsp;&nbsp;&nbsp;&nbsp;&bull; This can be assessed using bedtools |
| <h4>Percent Agreement</h4> | Percentage of base call concordance in reads mapped at a designated position in the reference genome|
| <h4>Coverage</h4> | What percent of the reference sequence is covered by the reads that have been produced<br/> &bull; This metric is typically used in conjunction with depth<br/>|
| <h4>Percent Mapped Reads</h4> | Percentage of read data mapped to a specified reference genome|
| <h4>Average Base Quality of Aligned Reads</h4> | Mean phred score of read data mapped to a reference genome|

### Consensus Assembly QC Metrics
An examination of the resulting assembly quality is also critical as these assemblies often inform critical downstream analysis, such as lineage and clade assignments and genomic epidmiology investigations. 

| Term                  | Definition                             |
| ---------------------- | --------------------------------------- |
| <h4>Length of the Assembly</h4> | Should be similar to that of reference. If it is not, why? Have there been large insertions/deletions, gene duplications, etc.| 
| <h4>Total Number of N’s</h4> | The total number of ambiguous basecalls in the assembly |
| <h4>Length of Strings of N’s</h4> | While the total number of N’s is important, the length of the strings of N’s can indicate issues with upstream laboratory workflows. If a string of N’s is consistently reported over a specific region of the genome, then one can cross reference the primer binding loci in the bed file to see if one amplicon is dropping out or amplifying at a lower rate than the other amplicons. This could be due to amplification bias, resulting from a large differential in the GC content between the amplicons. This may also indicate that you have a mixed population and there may be a subpopulation with a different sequence in the ambiguous region.|
| <h4>Percent Reference Coverage</h4> | Percentage of the Wuhan-1 reference genome represented in the consensus assembly|
| <h4>Number of Ns</h4> | Number of ambiguous base calls (Ns) incorporated into the consensus assembly|
| <h4>Assembly Length Unambiguous</h4> | Number of unambiguous base calls (ATCGs) incorporated into the consensus assembly|
| <h4>NTC Percent Coverage</h4> | Percentage of the Wuhan-1 reference genome represented in the consensus assembly of a non-template control (NTC; i.e. negative control)|
| <h4>Lineage Defining Mutations</h4> | Percentage of lineage-specific mutations represented in the consensus assembly|
| <h4>Number of Ns</h4> | Number of ambiguous base calls (Ns) incorporated into the consensus assembly|
| <h4>S-gene Coverage</h4> | Percentage of the SARS-CoV-2 S-gene represented in the consensus assembly|
| <h4>S-gene Frameshifts</h4> | S-gene insertion or deletion events represented in the consensus assembly|
| <h4>S-gene Ambiguous Bases</h4> | Number of ambiguous base calls (Ns) incorporated into the s-gene of the consensus assembly|


## Additional QC Resources and Materials

- [ncov-tools](https://github.com/jts/ncov-tools) - Tools and plots for performing quality control on coronavirus sequencing results.
- [Quality Management Systems Tools & Resources - Process Management](https://www.cdc.gov/labquality/qms-tools-and-resources.html#:~:text=Click%20to%20expand-,Process%20Management,-Provides%20guidance%20on) - US CDC Quality Management Systems for SARS-CoV-2 NGS Data
- [TheiaCoV QC output Video](https://www.youtube.com/watch?v=Amb-8M71umw&list=PLU47xRg_MKJrtyoFwqGiywl7lQj6vq8Uz&index=3) - Video tutorial for assessing SARS-CoV-2 genomic characterization with Theiagen's TheiaCoV workflows
- [StaPH-B Glossary](http://www.staphb.org/resources/glossary/) - US State Public Health Bioinformatics (StaPH-B) working group's bioinformatics glossary of terms
- [PHA4GE Bioinformatics Solutions](https://github.com/pha4ge/pipeline-resources/blob/main/docs/bioinfo-solutions.md) - This working groups list of bioinformatics solutions for SARS-CoV-2 bioinformatics 
- [ECDC: Guidance for representative and targeted genomic SARS-CoV-2 monitoring](https://www.ecdc.europa.eu/sites/default/files/documents/Guidance-for-representative-and-targeted-genomic-SARS-CoV-2-monitoring.pdf) - European CDC Guidance Document for SARS-CoV-2 genomic analysis
